src/uint256.h

```diff
@@ -21,72 +22,56 @@ class base_blob
 {
 protected:
     static constexpr int WIDTH = BITS / 8;
-    uint8_t m_data[WIDTH];
+    std::array<uint8_t, WIDTH> m_data;
+    static_assert(WIDTH == sizeof(m_data), "Sanity check");
 public:
     /* construct 0 value by default */
     constexpr base_blob() : m_data() {}
 
     /* constructor for constants between 1 and 255 */
     constexpr explicit base_blob(uint8_t v) : m_data{v} {}
 
-    explicit base_blob(const std::vector<unsigned char>& vch);
+    constexpr base_blob(Span<const unsigned char> vch)
+    {
+        assert(vch.size() == WIDTH);
+        std::copy(vch.begin(), vch.end(), m_data.begin());
+    }
 
-    bool IsNull() const
+    constexpr bool IsNull() const
     {
-        for (int i = 0; i < WIDTH; i++)
-            if (m_data[i] != 0)
-                return false;
-        return true;
+        return std::all_of(m_data.begin(), m_data.end(), [](uint8_t val){
```

Comment:
[`clang-format`](https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy) insists on:
```suggestion
        return std::all_of(m_data.begin(), m_data.end(), [](uint8_t val) {
```